model/database_functions.ts

```diff
@@ -59,6 +55,16 @@ export const createTrip = async (
 ): Promise<Trip> => {
   return database.write(async () => {
     const journey = await database.get<Journey>("journeys").find(journeyId);
+    const trips = await getAllTripsByJourneyId(journeyId);
+
+    // If this is the first trip, set the journey start date to today
+    if (trips.length === 0) {
+      const journey = await database.get<Journey>("journeys").find(journeyId);
+      await journey.update(() => {
+        journey.startedAt = Date.now();
+      });
+    }
+
```

Comment:
Ich weiß, ist ein Edge Case, aber: Wenn ich bereits eine/mehrere Strecken (Etappen) habe, dann aber eine manuell erstelle, die ein früheres Datum hat, dann steht da trotzdem das spätere Datum als Start der Reise (Tour).

Vielleicht fügst du noch ein else if hinzu, das etwa so aussieht (Achtung, Pseudocode):

````
createTrip(journeyId, title, tripStartedAt?) {      // tripStartedAt als optionaler Parameter fürs erstellen von manuellen Strecken

[...]

else if(tripStartedAt && tripStartedAt < journey.startedAt) {
    journey.startedAt = tripStartedAt;
}
````

Reply:
Nochmal hierzu. 

Ich bin mir nicht sicher, ob "startedAt" für dich der Anfang der Reise oder der Strecke ist. Für mich bedeutet es die Strecke. In dem Fall: Wenn ich eine manuelle Strecke eintrage, gebe ich ja jedes Mal eine Startzeit in diese Methode rein. Diese würde mit dem jetzigen Code *immer* das Startdatum der Reise mitsetzen, was natürlich problematisch ist, wenn bereits frühere Strecken existieren (also quasi immer). 
Es ist also sehr wichtig, vorher einen Datumsvergleich zu machen. Und "startedAt" natürlich auch als Startzeit des Trips setzen statt Date.now()

Falls du "startedAt" als Anfang der Reise interpretiert hast, finde ich das für diese Methode unpassend. Wenn ich eine manuelle Strecke erstelle, will ich nicht erst checken müssen, ob der Streckenstart das neue Startdatum der Reise ist. 

also hier mein Vorschlag für den Code (diesmal kein Pseudocode mehr)


```
export const createTrip = async (
  journeyId: string,
  title: string,
  startedAt?: number,
): Promise<Trip> => {
  return database.write(async () => {
    const journey = await database.get<Journey>("journeys").find(journeyId);
    const trips = await getAllTripsByJourneyId(journeyId);

    // If this is the first trip, set the journey start date to today
    if (trips.length === 0) {
      const journey = await database.get<Journey>("journeys").find(journeyId);
      await journey.update(() => {
        journey.startedAt = Date.now();
      });
    // If the trip's start time is earlier than the journey's start time, overwrite journey start Time
    } else if (startedAt && startedAt < journey.startedAt) {   
      await journey.update(() => {
        journey.startedAt = startedAt;
      });
   }

    return database.get<Trip>("trips").create((trip) => {
      trip.journey.set(journey);
      trip.title = title;
      trip.isActive = false;
      trip.startedAt = startedAt ?? Date.now();
    });
  });
```

Reply:
Weiß nicht, ob ich dich vielleicht falsch verstehe, aber das Datum wird doch eh nur gesetzt, wenn noch kein Trip vorhanden ist. Siehe `if (trips.length === 0)`.
Oder meinst du, wenn alle Trips gelöscht wurden? Sollte dann das Datum nicht auch neu gesetzt werden?

Ich überlege aber eh, ob man nicht einfach immer das früheste Startdatum aus allen Strecken nimmt und das dann irgendwie anders aufbaut. Vielleicht geht das auch mit diesen MySQL-Triggern? @SimonJohannesEngelhardt